context.go

```diff
@@ -19,58 +19,34 @@ import (
 
 // Ctx defines the supporting context of the tool.
 type Ctx struct {
+	WorkingDir string // Where to execute
+	Out, Err   *log.Logger
+	Verbose    bool     // Enables more verbose logging
 	GOPATH     string   // Selected Go path
 	GOPATHS    []string // Other Go paths
-	WorkingDir string
-	*Loggers
 }
 
-// Loggers holds standard loggers and a verbosity flag.
-type Loggers struct {
-	Out, Err *log.Logger
-	// Whether verbose logging is enabled.
-	Verbose bool
-}
-
-// NewContext creates a struct with the project's GOPATH. It assumes
+// SetGOPATH sets the project's GOPATH from an environment var. It assumes
 // that of your "GOPATH"'s we want the one we are currently in.
-func NewContext(wd string, env []string, loggers *Loggers) (*Ctx, error) {
-	ctx := &Ctx{WorkingDir: wd, Loggers: loggers}
-
-	GOPATH := getEnv(env, "GOPATH")
-	if GOPATH == "" {
-		GOPATH = defaultGOPATH()
+func (c *Ctx) SetGOPATH(gopath string) error {
```

Comment:
It seemed like it was left in an unusual form as an artifact of old behavior, with the `GOPATH` logic being the only real work left. In particular, having parameters to pass values straight through to exported fields on the returned struct seems like odd boilerplate - and it was going to compound without `Loggers`.

Would a better name (`Init`), or better documentation make usage more clear?  Alternatively, the `GOPATH` portion I've isolated could be encapsulated in a helper function (`goodFnName(gopath, wd string) (gopath, gopaths string, err error)`), so that `NewContext` can just set fields.